File: backend/src/main/java/com/yuma/app/document/Ingredients.java
package com.yuma.app.document;

import java.util.HashSet;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import org.springframework.data.mongodb.core.mapping.Document;

import com.yuma.app.document.enums.Allergens;
import com.yuma.app.document.enums.HealthLabels;

@Document
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Ingredients {

	private String name;
	private double weight;
	private double calories;
	private double price;
	private boolean optional;
	private HashSet<HealthLabels> healthLabels;
	private HashSet<Allergens> allergens;

	@Override
	public String toString() {
		return "Ingredients{" +
			"name='" + name + '\'' +
			", weight=" + weight +
			", calories=" + calories +
			", price=" + price +
			", optional=" + optional +
			'}';
	}
}

File: frontend/src/middleware/objects/Ingredients.js
class Ingredients { 

    constructor() {
      this.name = null;
      this.weight = null;
      this.calories = null;
      this.price = null;
      this.optional = null;
      this.allergens = null;
      this.healthLabels = null;
    }

    setName(name){this.name = name}
    setWeight(weight){this.weight = weight}
    setCalories(calories){this.calories = calories}
    setPrice(price){this.price = price}
    setOptional(optional){this.optional = optional}
    setAllergens(allergens){this.allergens = allergens}
    setHealthLabels(healthLabels){this.healthLabels = healthLabels}
  
    toString() {
      return (this.name + this.weight + this.calories + this.price + this.optional);
    }
}
export default Ingredients;
